addons/account/models/account_move_line.py

```diff
@@ -1914,6 +1920,12 @@ def _prepare_reconciliation_single_partial(self, debit_values, credit_values, sh
         credit_currency = credit_aml._get_reconciliation_aml_field_value('currency_id', shadowed_aml_values)
         company_currency = debit_aml.company_currency_id
 
+        is_cross_currency_payment = (
+            debit_currency != credit_currency
+            and debit_currency != company_currency
+            and credit_currency != company_currency
```

Comment:
Hello, there are chances that even if the debit and credit currencies are different, one of them (either debit or credit currency) can be same as the company currency, so, we need to involve company currency to make sure it's cross currency payment.

Reply:
Okay so i understand but i'm just making sure, let's say company currency is USD, a USD to EUR transaction is not cross_currency because one of them is the company currency ? Cross currency requires BOTH currency to be different from the company's ?

Reply:
Yes, in your case, when your company currency is USD and you are dealing with EUR, that's a foreign currency transaction, where single currency other than company currency is involved. In case when you have two currencies other than company currency, then we have a cross currency exchange / transaction.